### ISSUE TITLE
How to only perform an evaluation on custom dataset after loading a checkpoint from custom training?

### ISSUE THREAD

**Issue by Penaplion** [open-mmlab/mmdetection#8286]
Hello,
i just want to load a checkpoint and perform an evaluation on my validation set to get the bbox- and mask-metrics.
Is there a way to do that?

i've tried the following to achieve this:

evaluation = dict(interval=1, metric=['bbox', 'segm'])
runner = dict(max_epochs=1)

in combination with different workflows:
workflow = [('train', 0), ('val', 0)]
workflow = [('train', 0), ('val', 1)]
workflow = [('val', 1)]
workflow = [('train', 0)]
workflow = []

 but all not working, meaning it leads to endless loops or errors.

Thanks in advance!

**Comment by RangiLyu** [comment]
You can refer to the doc https://mmdetection.readthedocs.io/en/latest/2_new_data_model.html#test-and-inference
You need to run `tools/test.py` instead of `tools/train.py`

**Comment by Penaplion** [comment]
Thanks for response. Yup i already did some runs on my test set with several checkpoints as you mentioned above. But i was looking for the regular "evaluation phase" like you can set in the config: evaluation = dict(interval=..., metric=[...]).
E.g. when when i train a model for 10 epochs with this:
evaluation = dict(interval=10, metric=['bbox', 'segm'])
runner = dict(max_epochs=10)
workflow = [('train', 1), ('val', 1)]
checkpoint_config = dict(interval=1)

and in a plot of validation loss and train loss i see that the model overfits after epoch 6. Now i would like to roll back to checkpoint 6 (which is the epoch before overfitting starts) and perform only the evaluation on the validation set which was performed once, after 10 training epochs, before. I dont want to train again. Just do the evaluation because i would like to get the evaluation metrics at the state of epoch 6 but what i have are the evaluation metrics after 10 training epochs.

**Comment by RangiLyu** [comment]
You need to load the checkpoint `epoch_6.pth` instead of the latest checkpoint.
